### ISSUE TITLE
Swipe to close support

### ISSUE THREAD

**Issue by Dictory** [ascoders/react-native-image-viewer#45]
Hello @ascoders. I want say thanks for that lib. 
Can you add "swipe to close" future?
I think that useful future.

**Comment by ascoders** [comment]
Yes, it's useful future, i'll finish it when i'm free, also welcome pr :P .

**Comment by Dictory** [comment]
Ok, may be i can do that on weekend

**Comment by Dictory** [comment]
ok, I will try on this week. xD

**Comment by itswaze** [comment]
any update? @Dictory

**Comment by Dictory** [comment]
@itswaze sorry, I was so busy, I remember that. I try do in it soon.

**Comment by Dictory** [comment]
@ascoders hi, i have some trouble with import local version react-native-image-pan-zoom

```console
$ rm -rf built && tsc
node_modules/@types/react-native/index.d.ts(8810,11): error TS2451: Cannot redeclare block-scoped variable 'global'.
```
What i do wrong?

package: 
`"react-native-image-pan-zoom": "file:../react-native-image-zoom/"`

P.S. local version image-viewer work nice with test app

**Comment by ascoders** [comment]
![image](https://cloud.githubusercontent.com/assets/7970947/26713664/5df68d98-47a0-11e7-832c-deffe9b8b06b.png)

I did not find any errors when running locally. Use `yarn` install and retry?

**Comment by Dictory** [comment]
@ascoders, running locally work nice, try use local `react-native-image-pan-zoom` inside `image-viewer`.
With `react-native-image-pan-zoom` package from npm all work, from local repo i had error. But `react-native-image-pan-zoom` compile local without error. 
![2017-06-02 13 39 23](https://cloud.githubusercontent.com/assets/13243379/26713901/0ef7475c-4799-11e7-8c7d-a0ababec515b.png)
![2017-06-02 13 39 49](https://cloud.githubusercontent.com/assets/13243379/26713905/10b24fb0-4799-11e7-9cf5-6d22f0c9fc9b.png)

**Comment by ascoders** [comment]
Because of the duplicate copy of `@types/react-native`.

You can modify the `react-native-image-pan-zoom` source directly in node_modules and run `tsc` listening in its directory.

**Comment by Dictory** [comment]
@ascoders i knew what need do, but i can\`t dev it because i can\`t update packages local. I spent many time for it.

--------

Main idea: detect image move in PanResponderInstance onPanResponderMove (image-zoom.component.ts ln 161) to top or bottom when image doesn\`t zoom and use onDragTop and onDragBottom (props of image-zoom from image-viewer). 

--------

Do it without living test impossible for me. Sorry.

P.S. i linked image-zoom into image-viewer - that work, but i cant realtime updating image-viewer in project to actual.

**Comment by ascoders** [comment]
@Dictory Hard work, but I'm too busy working now, I really can't afford the time to realize this function.

The reason you can't realtime updating image-viewer is, you must run `tsc -w` in it's root path, because typescript needs to be converted to JavaScript.

You can install `react-native-image-pan-viewer`, find `react-native-image-pan-viewer` and `react-native-image-pan-zoom` in node_modules, run `tsc -w` in their root, then modify the source code will take effect immediately.

**Comment by tikkichan4** [comment]
It would be a perfect library for me if the swipe to close function is implemented. Looking forward to the great works of you guys !

**Comment by ragnorc** [comment]
+1 Me too!

**Comment by niocncn** [comment]
any updates here?

**Comment by ShayanJavadi** [comment]
@Dictory are you still working on this? I can pick this up and implement it if you're not.

**Comment by Dictory** [comment]
@ShayanJavadi will be awesome. Now i work other RNless project and doesn`t have time

**Comment by ShayanJavadi** [comment]
@Dictory I'll get started on it then

**Comment by ShayanJavadi** [comment]
it's done. about to make a pr soon

**Comment by sfatih** [comment]
wow @ShayanJavadi and @ascoders i was looking for this today. And i saw 2.1.0 version. Thank you very much

**Comment by berpcor** [comment]
And why only swipe down? If you want to close image, are you really intuitevely swipe down??
Anyway, please add swipe up to close. And good practice is change opacity of background while swiping down or up.

**Comment by ascoders** [comment]
@berpcor It's interesting.

**Comment by davidsneal** [comment]
@ascoders I'd really like swipe up too please if you can :)

**Comment by suretarget** [comment]
Hi there, I really like this package. 
Anyone can implement swipe up to close? Workaround for temporary?

**Comment by davidsneal** [comment]
@ascoders I am more than happy to put together a PR to add swipe up to close support. I have tried to understand how I would contribute but had no joy. Do you have any contribution guidelines please?

**Comment by Egizas** [comment]
Hey guys, is there any update on **swipe up to close** feature?

**Comment by tom-krusch** [comment]
+1 swipe up to close
